File: app/src/main/java/com/boardgamegeek/auth/AuthenticationService.java
package com.boardgamegeek.auth;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

import timber.log.Timber;

public class AuthenticationService extends Service {
	private Authenticator authenticator;

	@Override
	public void onCreate() {
		Timber.v("BoardGameGeek Authentication Service started.");
		authenticator = new Authenticator(this);
	}

	@Override
	public void onDestroy() {
		Timber.v("BoardGameGeek Authentication Service stopped.");
	}

	@Override
	public IBinder onBind(Intent intent) {
		Timber.v("getBinder: returning the AccountAuthenticator binder for intent " + intent);
		return authenticator.getIBinder();
	}
}

File: app/src/main/java/com/boardgamegeek/auth/AuthenticationService.kt
package com.boardgamegeek.auth

import android.app.Service
import timber.log.Timber
import android.content.Intent
import android.os.IBinder
import com.boardgamegeek.repository.AuthRepository
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class AuthenticationService : Service() {
    @Inject lateinit var authRepository: AuthRepository
    private var authenticator: Authenticator? = null

    override fun onCreate() {
        super.onCreate()
        Timber.v("BoardGameGeek Authentication Service started.")
        authenticator = Authenticator(this, authRepository)
    }

    override fun onDestroy() {
        Timber.v("BoardGameGeek Authentication Service stopped.")
    }

    override fun onBind(intent: Intent): IBinder? {
        Timber.v("getBinder: returning the AccountAuthenticator binder for intent %s", intent.toString())
        return authenticator?.iBinder
    }
}
